store/catalog.go

```diff
@@ -19,28 +19,13 @@ var (
 
 // Catalog is the database catalog.
 type Catalog struct {
-	databaseID *int
-	store      *Store
-	engineType db.Type
+	Database *catalog.Database
 }
 
 // NewCatalog creates a new database catalog.
-func NewCatalog(databaseID *int, store *Store, dbType db.Type) *Catalog {
-	return &Catalog{
-		databaseID: databaseID,
-		store:      store,
-		engineType: dbType,
-	}
-}
-
-// GetDatabase implements the catalog.Catalog interface.
-func (c *Catalog) GetDatabase(ctx context.Context) (*catalog.Database, error) {
-	if c.databaseID == nil {
-		return &catalog.Database{}, nil
-	}
-
-	database, err := c.store.GetDatabase(ctx, &api.DatabaseFind{
-		ID: c.databaseID,
+func NewCatalog(ctx context.Context, databaseID int, store *Store, engineType db.Type) (*Catalog, error) {
```

Comment:
This should return catalog.Catalog, error

*Catalog implements catalog.Catalog which is an implementation detail

Reply:
Nice catch